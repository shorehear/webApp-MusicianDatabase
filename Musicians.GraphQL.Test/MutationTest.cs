using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Musicians.Database;
using CookieCrumble;

namespace Musicians.Test
{
    public class MutationTest : IClassFixture<TestService>
    {
        private readonly IServiceProvider serviceProvider;

        public MutationTest(TestService testService)
        {
            serviceProvider = testService.ServiceProvider;
        }

        [Fact]
        public async Task TestCreateMusician()
        {
            using var scope = serviceProvider.CreateScope();
            var contextFactory = scope.ServiceProvider.GetRequiredService<IDbContextFactory<MusiciansDbContext>>();
            using var context = contextFactory.CreateDbContext();

            SeedTestData.Initialize(context);

            var executor = await scope.ServiceProvider.GetRequiredService<IRequestExecutorResolver>()
                .GetRequestExecutorAsync();

            var mutation = ("""
                mutation {
                  createMusician(musicianName: "Freddie Mercury",
                   musicianBirthDate: "05.09.1946",
                    countryName: "UK") {
                    id
                    musicianBirthDate
                    musicianName
                  }
                }
                """);

            var result = await executor.ExecuteAsync(mutation);

            result.MatchSnapshot();
        }

        [Fact]
        public async Task TestRegisterAndLoginUser()
        {
            using var scope = serviceProvider.CreateScope();
            var contextFactory = scope.ServiceProvider.GetRequiredService<IDbContextFactory<MusiciansDbContext>>();
            using var context = contextFactory.CreateDbContext();

            SeedTestData.Initialize(context);

            var executor = await scope.ServiceProvider.GetRequiredService<IRequestExecutorResolver>()
                .GetRequestExecutorAsync();

            var registerMutation = """
                mutation {
                  registerUser(username: "testuser", password: "password123") {
                    id
                    username
                  }
                }
                """;

            var registerResult = await executor.ExecuteAsync(registerMutation);
            registerResult.MatchSnapshot("RegisterUserSnapshot");

            var loginMutation = """
            mutation {
              login(username: "testuser", password: "password123") 
            }
            """;

            var loginResult = await executor.ExecuteAsync(loginMutation);
            loginResult.MatchSnapshot("LoginUserSnapshot");
        }
    }
}
