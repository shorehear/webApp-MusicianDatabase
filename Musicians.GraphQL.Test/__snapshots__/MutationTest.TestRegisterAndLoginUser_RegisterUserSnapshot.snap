{
  "data": {
    "registerUser": {
      "id": "f65fe7cb-e282-4079-98d6-83e001fe6b87",
      "username": "testuser"
    }
  }
}